react-common/components/controls/Feedback/FeedbackEventListener.ts

```diff
@@ -0,0 +1,158 @@
+        appId: appId,
+        ageGroup: ocv.FeedbackAgeGroup.Undefined,
+        authenticationType: ocv.FeedbackAuthenticationType.Unauthenticated,
+        clientName: "MakeCode",
+        feedbackConfig: feedbackConfig,
+        isProduction: false,
+        themeOptions: themeOptions,
+        // telemetry - will likely want this
+    }
+
+    FEEDBACK_FRAME_ID = frameId;
+}
+
+export const removeFeedbackEventListener = () => {
+    window.removeEventListener('message', feedbackCallbackEventListener);
+}
+
+/**
+ * The function that listens for the feedback events.
+ * The events here are the ones that seemed most useful to log or respond to
+ * @param event: the event received from OCV
+ */
+const feedbackCallbackEventListener = (event: MessageEvent<FeedbackRequestPayloadType>) => {
+    if (event.data.Event) {
+        const payload: FeedbackRequestPayloadType = event.data
+        switch (payload.Event) {
+            case 'InAppFeedbackInitOptions': //This is required to initialise feedback
+                sendFeedbackInitOptions()
+                break
+            case 'InAppFeedbackOnError': //Invoked when an error occurrs on feedback submission - would be nice to log something to the user
+                pxt.warn('Error Message: ', payload.EventArgs)
+                break
+            case 'InAppFeedbackInitializationComplete': //Invoked when feedback form is fully initialised and displays error/warning if any - nice to have a log for this
+                pxt.debug('InAppFeedbackInitializationComplete: ', payload.EventArgs)
+                break
+            case 'InAppFeedbackOnSuccess': //Invoked when feedback submission is successful - would be useful to have telemetry/something else on this event
+                pxt.debug('InAppFeedbackOnSuccess: ', payload.EventArgs)
+                break
+            case 'InAppFeedbackDismissWithResult': //Invoked when feedback is dismissed - the big important one for us to be able to close the feedback modal
+                pxt.debug('InAppFeedbackDismissWithResult: ', payload.EventArgs);
+                if (feedbackCallbacks.onDismiss) {
+                    feedbackCallbacks.onDismiss();
+                }
+                break
+        }
+    }
+}
+
+// ***************** Helper Functions *****************
+
+// TODO
+// haven't implemented yet with events, but this will be needed in order to update to high contrast
+// general changes need to be made as well use the correct theme. the windows ones were just the defaults.
+const sendUpdateTheme = () => {
+    if (currentTheme == 'WindowsDark') {
+        currentTheme = 'WindowsLight'
+    } else {
+        currentTheme = 'WindowsDark'
+    }
+    const response: FeedbackResponsePayloadType = {
+        event: 'OnFeedbackHostAppThemeChanged',
+        data: {
+            baseTheme: currentTheme,
+        },
+    }
+    const iFrameElement = document.getElementById(FEEDBACK_FRAME_ID) as HTMLIFrameElement
```

Comment:
Code duplication here. Move these two lines to a helper function. In that function, check for missing iFrameElement, as document.getElementById can return undefined.